src/compiler/checker.ts

```diff
@@ -6824,41 +6824,52 @@ namespace ts {
                 getSignaturesOfType(type, SignatureKind.Construct).length === 0;
         }
 
+        function createTransientSymbol(source: Symbol, type: Type) {
+            const symbol = <TransientSymbol>createSymbol(source.flags | SymbolFlags.Transient, source.name);
+            symbol.declarations = source.declarations;
+            symbol.parent = source.parent;
+            symbol.type = type;
+            symbol.target = source;
+            if (source.valueDeclaration) symbol.valueDeclaration = source.valueDeclaration;

```

Comment:
I'll change it here, but it's written this way in a couple of other symbol-creating functions.